src/raglite/_embed.py

```diff
@@ -137,50 +137,40 @@ def _create_segment(
     return sentence_embeddings
 
 
+def _embed_string_batch(string_batch: list[str], *, config: RAGLiteConfig) -> FloatMatrix:
+    """Embed a batch of text strings."""
+    if config.embedder.startswith("llama-cpp-python"):
+        # LiteLLM doesn't yet support registering a custom embedder, so we handle it here.
+        # Additionally, we explicitly manually pool the token embeddings to obtain sentence
+        # embeddings because token embeddings are universally supported, while sequence
+        # embeddings are only supported by some models.
+        embedder = LlamaCppPythonLLM.llm(
+            config.embedder, embedding=True, pooling_type=LLAMA_POOLING_TYPE_NONE
+        )
+        embeddings = np.asarray([np.mean(row, axis=0) for row in embedder.embed(string_batch)])
+    else:
+        # Use LiteLLM's API to embed the batch of strings.
+        response = embedding(config.embedder, string_batch)
+        embeddings = np.asarray([item["embedding"] for item in response["data"]])
+    # Normalise the embeddings to unit norm and cast to half precision.
+    if config.embedder_normalize:
+        embeddings /= np.linalg.norm(embeddings, axis=1, keepdims=True)
```

Comment:
Not related to this PR per se, but this is a bit safer (I've seen all-zero embeddings before, for example of the empty string):
```suggestion
        eps = np.finfo(embeddings.dtype).eps
        norm = np.linalg.norm(embeddings, axis=1, keepdims=True)
        embeddings /= np.maximum(norm, eps)
```